plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -180,6 +189,37 @@ protected Transaction newCreateTableTransaction(
         return createTableTransaction(schemaTableName.toString(), ops, metadata);
     }
 
+    protected abstract boolean schemaExists(SchemaTableName schemaTableName);
```

Comment:
in TrinoCatalog interface, these things are called "namespaces"

also, `SchemaTableName` is not a good identifier for a "namespace" nor "schema"
did you intend to pass a String here?

Reply:
BTW you can have a default implementation for this method, using `listNamespaces`.

This isn't optimal (https://github.com/trinodb/trino/pull/14375#discussion_r985699629), but i see we're already using this for `ConnectorMetadata.schemaExists` implementation in Iceberg (ie we ride on the default implementation).
This is a room for improvement, but `ConnectorMetadata.schemaExists` is definitely a more important one to improve, compared with the code invoked within `register_table` procedure only.
Thus

- let's use `listNamespaces` for now
- with a TODO comment to improve this when we improve `ConnectorMetadata.schemaExists` for Iceberg

Reply:
I am adding `namespaceExists` in https://github.com/trinodb/trino/pull/14457